src/components/element-editor/basic-details/preselected-value.tsx

```diff
@@ -0,0 +1,129 @@
+import { Form } from "@lifesg/react-design-system/form";
+import { Text } from "@lifesg/react-design-system/text";
+import { useEffect } from "react";
+import { Controller, useFormContext } from "react-hook-form";
+import {
+    EElementType,
+    IDropdownItemAttributes,
+} from "src/context-providers/builder/";
+import { TOverallOptionGroupBasedSchema } from "src/schemas";
+
+export const PreselectedValue = () => {
+    // ===========================================================================
+    // CONST, STATE, REFS
+    // ===========================================================================
+    const {
+        control,
+        formState: { errors },
+        watch,
+        setValue,
+    } = useFormContext<TOverallOptionGroupBasedSchema>();
+
+    const type = watch("type");
+    const preselectedValue = watch("preselectedValue");
+
+    useEffect(() => {
+        if (preselectedValue === "") {
+            setValue("preselectedValue", null, {
+                shouldDirty: true,
+            });
+        }
+    }, [preselectedValue, setValue]);
+
+    // =========================================================================
+    // RENDER FUNCTIONS
+    // =========================================================================
+
+    const renderController = () => {
+        switch (type) {
+            case EElementType.DROPDOWN:
+                const dropdownItems = watch("dropdownItems");
+                const preselectedValueOptions: IDropdownItemAttributes[] = [
+                    {
+                        label: "Unselect default value",
+                        value: "",
+                    },
+                ];
+
+                const validDropdownItems = dropdownItems?.filter(
+                    (item): item is IDropdownItemAttributes => {
+                        return item.value.length > 0 && item.label.length > 0;
+                    }
+                );
+
+                if (validDropdownItems?.length > 0) {
+                    preselectedValueOptions.push(...validDropdownItems);
+                }
+
+                if (
+                    preselectedValue &&
+                    !validDropdownItems?.some(
+                        (item) => item.value === preselectedValue
+                    )
+                ) {
+                    setValue("preselectedValue", null);
+                }
```

Comment:
- combine filter and push in a single `forEach` function
- use `isEmpty` to check for empty strings
- can do without `validDropdownItems`

```suggestion
                dropdownItems.forEach((item: IDropdownItemAttributes) => {
                    if (!isEmpty(item.label) && !isEmpty(item.value)) {
                        preselectedValueOptions.push(item);
                    }
                });
                if (
                    preselectedValue &&
                    !dropdownItems.some(
                        (item) => item.value === preselectedValue
                    )
                ) {
                    setValue("preselectedValue", null);
                }
```